crates/parser/src/lib.rs

```diff
@@ -155,7 +161,10 @@ fn parse_lines_count_match_pattern_rule(
     Ok((name, match_type, pattern, comparison, timeout))
 }
 
-fn parse_custom_script_rule(record: Pair<Rule>, is_js: bool) -> Result<AssertionKind, ParserError> {
+fn parse_custom_script_rule(
+    record: Pair<Rule>,
+    script_type: ScriptType,
+) -> Result<AssertionKind, ParserError> {
```

Comment:
I think all these methods should be implemented on Pair<Rule>, for example:

```rust
impl Pair<Rule> {
     fn parse_custom_script_rule(&self, script_type: ScriptType) -> Result<AssertionKind, ParserError> {
       todo!()
     }
}
```

Then you can make the code cleaner:
```rust
let record = Pair {};

record.parse_custom_script_rule()?;
```

Reply:
We can do that in a different PR

Reply:
Maybe just add a #[TODO] ?

Reply:
Even better :) I'll add an issue 😉